mlflow/tracking/fluent.py

```diff
@@ -302,8 +297,21 @@ def _get_experiment_id_from_env():
     return env.get_env(_EXPERIMENT_ID_ENV_VAR)
 
 
+def _get_or_create_experiment(experiment_name):
+    client = MlflowClient()
+    exp = client.get_experiment_by_name(experiment_name)
+    exp_id = exp.experiment_id if exp else None
+    if exp_id is None:
+        print("INFO: '{}' does not exist. Creating a new experiment".format(experiment_name))
+        client.create_experiment(experiment_name)
+        return client.get_experiment_by_name(experiment_name)
+    return exp
+
+
 def _get_experiment_id():
-    return int(_active_experiment_id or
-               _get_experiment_id_from_env() or
-               (is_in_databricks_notebook() and get_notebook_id()) or
-               Experiment.DEFAULT_EXPERIMENT_ID)
+    exp_id = int(_active_experiment_id or _get_experiment_id_from_env() or
+                 (is_in_databricks_notebook() and get_notebook_id()))
+    if not exp_id:
```

Comment:
Do you mean `exp_id is None`. In any case, this changes in the intension of `_get_experiment_id` method. This is left open in case, users do not want to have a default experiment in their system.